Add option to generate all paths relative to root.

tritao opened the pull request:
Depends on: https://github.com/jimon/premake-ninja/pull/46

See only https://github.com/jimon/premake-ninja/commit/8dad225dc9531ed91876ad703d9534eb9354b98f for review.

Add option to generate all paths relative to root. This adds a new generation mode that keeps all paths relative to the
root script dir.

This forces GCC to write paths in diagnostics relative to the root script, which allows VS Code to recognize the paths when ctrl-clicking the paths in compiler diagnostics in terminal.

Without this, we get path relative to the build folder, `../../foo.c`, which VS Code cannot recognize and hinders instant code navigation.

Jarod42 commented:
[INFO]
> Depends on: https://github.com/jimon/premake-ninja/pull/46

Possible way is to target that branch (so current PR is retargeted once first PR is merged).
Not sure if it works cross repo though.

tritao committed:
Fix PCH regression due to bad scoping.

Originally introduced in 8047f16488b3b5bad28c6a48fc1adaaca9d727b1

tritao committed:
Add option to generate all paths relative to root.

This adds a new generation mode that keeps all paths relative to the
root script dir.

This forces GCC to write paths in diagnostics relative to the root
script, which allows VS Code to recognize the paths when ctrl-clicking
the paths in compiler diagnostics in terminal.

Without this, we get path relative to the root, `../../foo.c`, which VS
Code cannot recognize and hinders instant code navigation.
